### PR TITLE
Adds complexity assessment

### LINKED ISSUE #347: Add complex words assessment
We recommend limiting the use of complex words (i.e. words consisting of ≥4 syllables) to 10% of the total word count. We should calculate the percentage of words with ≥4 syllables, and come up with a score. To score the use of complex words, the following criteria should be met: 

Recommended maximum percentage of words ≥4 syllables = 10 

Calculate percentage of words ≥4 syllables:

```JS
complexWordsPercentage = complexWordCount / wordCount * 100
```

Calculate score:
```JS
9 - Math.max([Math.min([(6 / 10)*(complexWordsPercentage - 6.7), 6]), 0])
```

Add feedback:

score >= 7: “%1$d% of the words contain over %2$d syllables, which is within the recommended range.” 

score < 7: “%1$d% of the words contain over %2$d syllables, which is more than the recommended maximum of %3$d%.”

### PR DESCRIPTION
**Pull request by terw-dan**
Creates an assessment for the complexity of words. 

Fixes #347

For testing, run `grunt build` and test in the browserified example

### CODE REVIEW

File: js/assessments/wordComplexityAssessment.js [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+ * @param {number} recommendedValue The recommendedValue for amount of syllables.
+ * @param {object} i18n The object used for translations.
+ * @returns {{score: number, text}} resultobject with score and text.
+ */
+var calculateComplexity = function( wordCount, tooComplexWords, recommendedValue, i18n ) {
+	var percentage = ( tooComplexWords / wordCount ) * 100;
+	var recommendedMaximum = 10;
+	// 6 is the number of scorepoints between 3, minscore and 9, maxscore. For scoring we use 10 steps.
+	// Up to 6.7 percent is for scoring a 9, higher percentages give lower scores.
+	var score = 9 - Math.max( Math.min( ( 6 / 10 ) * ( percentage - 6.7 ), 6 ), 0 );
```
Comment (moorscode) [inline js/assessments/wordComplexityAssessment.js]: Optimisation: 0.6 instead of 6 / 10

File: js/assessments/wordComplexityAssessment.js [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+ * Execute the word complexity assessment and return a result based on the syllables in words
+ * @param {Paper} paper The Paper object to assess.
+ * @param {Researcher} researcher The Researcher object containing all available researches.
+ * @param {object} i18n The object used for translations
+ * @returns {AssessmentResult} The result of the assessment, containing both a score and a descriptive text.
+ */
+var wordComplexityAssessment = function( paper, researcher, i18n ) {
+	var wordComplexity = researcher.getResearch( "wordComplexity" );
+	var wordCount = wordComplexity.length;
+	var recommendedValue = 4;
```
Comment (moorscode) [inline js/assessments/wordComplexityAssessment.js]: I can imagine might provide false positives for languages other than English, German has a lot of long words which might not be very complex.
Can we set this from config?

File: js/assessments/wordComplexityAssessment.js [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+ * @param {object} i18n The object used for translations.
+ * @returns {{score: number, text}} resultobject with score and text.
+ */
+var calculateComplexity = function( wordCount, tooComplexWords, recommendedValue, i18n ) {
+	var percentage = ( tooComplexWords / wordCount ) * 100;
+	var recommendedMaximum = 10;
+	// 6 is the number of scorepoints between 3, minscore and 9, maxscore. For scoring we use 10 steps.
+	// Up to 6.7 percent is for scoring a 9, higher percentages give lower scores.
+	var score = 9 - Math.max( Math.min( ( 6 / 10 ) * ( percentage - 6.7 ), 6 ), 0 );
+	if ( score >= 7 ) {
```
Comment (moorscode) [inline js/assessments/wordComplexityAssessment.js]: Missing `fixFloatingPoint( score );`

File: js/assessments/wordComplexityAssessment.js [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+		return {
+			score: score,
+			// translators: %1$d expands to the number of too complex words, %2$d expands to the recommended number of syllables
```
Comment (jcomack) [inline js/assessments/wordComplexityAssessment.js]: I believe these translator strings need to be moved to be directly above the actual translate.

This also applies for the translator string on line 31.

File: js/assessments/wordComplexityAssessment.js (hunk 2 above)
Comment (terw-dan) [inline js/assessments/wordComplexityAssessment.js]: For now we will keep this at 4, untill we have an assessment that can handle other languages, and we know what we need to change.

**Comment by jcomack** [comment]
CR done 👍

**Comment by jcomack** [comment]
Acceptance done.

It seems like there's some rounding of percentages, missing (see screenshot).

yoast_seo_content_analysis

**Comment by IreneStr** [comment]
Acceptance done :+1:

**Pull request merged**